dev-libs/uriparser/uriparser-0.9.1~20190120.recipe

```diff
@@ -42,28 +49,54 @@ REQUIRES_devel="
 BUILD_REQUIRES="
 	haiku${secondaryArchSuffix}_devel
 	"
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+BUILD_REQUIRES="$BUILD_REQUIRES
+	devel:libgtest$secondaryArchSuffix
+	"
+fi
+
 BUILD_PREREQUIRES="
-	cmd:aclocal
-	cmd:autoconf
-	cmd:automake
-	cmd:autoreconf
+	cmd:cmake
 	cmd:gcc$secondaryArchSuffix
 	cmd:libtoolize$secondaryArchSuffix
 	cmd:make
 	"
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
+BUILD_PREREQUIRES="$BUILD_PREREQUIRES
+	cmd:dot
+	cmd:doxygen
+	cmd:pkg_config$secondaryArchSuffix
+	cmd:qhelpgenerator$secondaryArchSuffix
+	"
+fi
+
 
 defineDebugInfoPackage uriparser$secondaryArchSuffix \
 	"$libDir"/liburiparser.so.$libVersion
 
+if [ "$effectiveTargetArchitecture" != x86_gcc2 ]; then
```

Comment:
Instead of doing like so, we can use a simpler variant:

```sh
cmakeArgs=( $cmakeDirArgs )
if [ "$effectiveTargetArchitecture" = x86_gcc2 ]; then
  cmakeArgs+=( -DURIPARSER_BUILD_TESTS=OFF -DURIPARSER_BUILD_DOCS=OFF )
fi

cmake .. "${cmakeArgs[@]}"

Reply:
Maybe @hartwork could take a look here, perhaps a check for the compiler could be added to switch the option to OFF in the CMakeLists.txt file? but yes looks ok (not tested yet)

Reply:
GCC 2.x is EOL outside of Haiku so I welcome that you applied GCC 2.x specifics on Haiku package level now

Reply:
Or not at all... Is there an existing gcc2 app that depends on it and cannot be easily migrated to gcc7?

Reply:
Searching haikuports I see only boca using it: https://github.com/haikuports/haikuports/search?q=lib%3Aliburiparser&unscoped_q=lib%3Aliburiparser
Not sure if Haiku itself is using it?

Reply:
If not we could remove the gcc2  existence :)